lib/actions-oclif/push.ts

```diff
@@ -261,70 +185,50 @@ export const push: CommandDefinition<
 				Note that if the service name cannot be found in the composition, the entire
 				left hand side of the = character will be treated as the variable name.
 			`,
-		},
-		{
-			signature: 'convert-eol',
-			alias: 'l',
-			description: isV12()
-				? 'No-op and deprecated since balena CLI v12.0.0'
-				: stripIndent`
-				On Windows only, convert line endings from CRLF (Windows format) to LF (Unix format).
-				Source files are not modified.`,
-			boolean: true,
-		},
-		...(isV12()
-			? [
-					{
-						signature: 'noconvert-eol',
-						description:
-							"Don't convert line endings from CRLF (Windows format) to LF (Unix format).",
-						boolean: true,
-					},
-			  ]
-			: []),
-		{
-			signature: 'multi-dockerignore',
-			alias: 'm',
+			multiple: true,
+		}),
+		'convert-eol': flags.boolean({
+			description: 'No-op and deprecated since balena CLI v12.0.0',
+			char: 'l',
+			hidden: true,
+		}),
+		'noconvert-eol': flags.boolean({
+			description: `Don't convert line endings from CRLF (Windows format) to LF (Unix format).`,
+		}),
+		'multi-dockerignore': flags.boolean({
 			description:
 				'Have each service use its own .dockerignore file. See "balena help push".',
-			boolean: true,
-		},
-		{
-			signature: 'nogitignore',
-			alias: 'G',
+			char: 'm',
+			exclusive: ['gitignore'],
+		}),
+		nogitignore: flags.boolean({
 			description:
 				'No-op (default behavior) since balena CLI v12.0.0. See "balena help push".',
-			boolean: true,
-		},
-		{
-			signature: 'gitignore',
-			alias: 'g',
+			char: 'G',
+			hidden: true,
+		}),
+		gitignore: flags.boolean({
 			description: stripIndent`
 				Consider .gitignore files in addition to the .dockerignore file. This reverts
 				to the CLI v11 behavior/implementation (deprecated) if compatibility is required
 				until your project can be adapted.`,
-			boolean: true,
-		},
-	],
-	async action(params, options) {
+			char: 'g',
+			exclusive: ['multi-dockerignore'],
+		}),
+		help: cf.help,
+	};
+
+	public static primary = true;
+
+	public async run() {
+		const { args: params, flags: options } = this.parse<FlagsDef, ArgsDef>(
+			PushCmd,
+		);
+
 		const sdk = getBalenaSdk();
-		const Bluebird = await import('bluebird');
 		const remote = await import('../utils/remote-build');
 		const deviceDeploy = await import('../utils/device/deploy');
```

Comment:
Could you move this import down into the branch where it's actually used, that way we can avoid the import for remote builds :)